### ISSUE TITLE
Feature Request :: Add driver for Realtek 5GbE (8126) Network Interface

### ISSUE THREAD

**Issue by Grim-GH** [rescuezilla/rescuezilla#512]
Not sure if this is possible, but I'd love to be able to clone across the network, and the current version of Rescuezilla does not include the driver for my fairly new NIC (Realtek 5GbE (8126)). 

The Linux driver for Kernel 6.4+ can be found here: https://www.realtek.com/Download/List?cate_id=584

**Comment by shasheene** [comment]
Yep, drivers in newer Linux kernel coming with upcoming versions of Rescuezilla based on more recent versions of Ubuntu.

Ubuntu 24.04 comes (current Rescuezilla v2.5.1 "Noble" release) comes with (quick google) kernel version 6.8.0.

Ubuntu 24.10 (future Rescuezilla release based on Ubuntu 24.10 Oracular comes with (quick google) Linux 6.11 kernel.

Quick search also [reveals](https://www.neowin.net/news/ubuntu-2410-oracular-oriole-arrives-with-latest-kernel-and-gnome-47/): 

> Ubuntu 24.10 launches with the Linux 6.11 kernel. According to Canonical, _this marks a shift in its kernel selection policy so that the newest available upstream Linux kernel is chosen_ so that users get the latest features and hardware support at release. Doing this also benefits silicon vendors, who can now easily plan for future kernel versions for Ubuntu releases and hardware enablement updates to LTS releases.

(_italic emphasis_ mine)

I'll be releasing at least one new version of Rescuezilla closer to the end of the year.

**Comment by shasheene** [comment]
(Revisiting this old GitHub issues to try and close it)

OK, I can confirm from the Linux kernel source code that support for the RealTek RTL8126 chipset is available through the in-kernel driver named `r8169`:

```
$  git log --pretty=format:"%h %ad %s" --date=short drivers/net/ethernet/realtek/r8169_main.c | grep 8126
4dec0702b862 2025-04-18 r8169: merge chip versions 70 and 71 (RTL8126A)
b48688ea3c9a 2025-03-18 r8169: disable RTL8126 ZRX-DC timeout
473367a5ffe1 2025-03-07 r8169: increase max jumbo packet size on RTL8125/RTL8126
b299ea006928 2024-12-13 r8169: adjust version numbering for RTL8126
e3e9e9039fa6 2024-11-06 r8169: align WAKE_PHY handling with r8125/r8126 vendor drivers
2cd02f2fdd8a 2024-11-02 r8169: improve initialization of RSS registers on RTL8125/RTL8126
87e26448dbda 2024-10-09 r8169: don't apply UDP padding quirk on RTL8126A
3b067536daa4 2024-09-18 r8169: add missing MODULE_FIRMWARE entry for RTL8126A rev.b
69cb89981c7a 2024-08-30 r8169: add support for RTL8126A rev.b
f4d3e595c000 2024-02-17 r8169: add MODULE_FIRMWARE entry for RTL8126A
be51ed104ba9 2024-02-12 r8169: add LED support for RTL8125/RTL8126
3907f1ffc0ec 2024-02-01 r8169: add support for RTL8126A
```

So concurring with my prior comment, Rescuezilla support is in-line with Linux kernel releases and the kernels available in the relevant variants of Rescuezilla.

But for completeness, while researching this to close this issue, I noticed there's actually some packages in the Ubuntu repository for the Realtek NIC drivers (using the "DKMS" method of compiling the source-code upon installation of the package rather than shipping a precompiled binary driver):

- https://packages.ubuntu.com/noble/r8125-dkms
- https://packages.ubuntu.com/noble/r8168-dkms
- https://packages.ubuntu.com/noble/rtl8812au-dkms

But viewing the files in those packages, the drivers for 8126 aren't included in those. I can separately see a third-party DKMS packaging of some drivers:

- https://github.com/awesometic/realtek-r8125-dkms
- **https://github.com/awesometic/realtek-r8126-dkms** (your one)
- https://github.com/awesometic/realtek-r8152-dkms

I am explicitly not including these packages because as best I can tell the in-kernel `r8169` driver provides everything we need. There is some discussion and debate here that may be relevant to future readers about different Realtek drivers: https://askubuntu.com/questions/1279644/r8168-r8169-which-one-should-i-use

As far as Rescuezilla goes, this issue is 'Done' with newer Rescuezilla releases, so I am closing this old GitHub issue.
